exercises/brandon_hewer/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,216 @@
+			'"', ' ', ';', '\t'};
+		auto word_cond = [](const string s) -> const bool {
+			return s.size() >= MIN_WORD_LENGTH;
+		};
+
+		// Retrieve words from input file.
+		vector<string> words = getWords(ifile, delims);
+		ifile.close();
+
+		// Count the number of occurrences of each unique word.
+		unordered_map<string, int> count = countUniqueWords(words, word_cond);
+
+		// Output the total occurrences of each unique word in a results file.
+		ofstream ofile("results.txt");
+		ofile << "Word\tUsage\n" << flip_map(count);
+		ofile.close();
+
+		return 1;
+	}
+}
+
+/*
+ * Adds the data contained in the specified multimap \p mp, to the
+ * specified output stream \p os.
+ *
+ * Each key-value entry in the multimap is added to the output stream
+ * in the format "KEY\tVALUE\n". Key-value entries are added in reverse
+ * order.
+ *
+ * \param os	The output stream to add to.
+ * \param mp	The multimap whose data to add to the output stream.
+ *
+ * \returns		The output stream, with the data from the output stream
+ *              added.
+ */
+ostream &operator<<(ostream &os, const multimap<int, string>& mp) {
+	multimap<int, string> mmp;
+	
+	// Iterates over each key-value entry in the specified map /p mp.
+	for (auto i = mp.rbegin(); i != mp.rend(); ++i) {
+		os << i->second << "\t" << i->first << "\n";
+	}
+	return os;
+}
+
+/*
+ * Generates a multimap whose pairs are created from flipping
+ * the key-value entries of the specified map \p mp.
+ *
+ * \param mp	The map whose key-value entires to flip.
+ *
+ * \returns		The multimap resulting from flipping the
+                key-value entries of the specified map.
+ */
+template<typename A, typename B>
+multimap<B, A> flip_map(const unordered_map<A, B>& mp) {
+	multimap<B, A> mmp;
+
+	// Iterates over each key-value entry in the specified
+	// map \p mp.
+	for (auto i = mp.begin(); i != mp.end(); ++i) {
+		mmp.insert(pair<B, A>(i->second, i->first));
+	}
+	return mmp;
+}
+
+/*
+ * Count the number of occurrences of each unique word in the
+ * specified vector of words \p words.
+ *
+ * \param words	The vector of words to count the number of
+ *              occurrences of each unique word from.
+ * \param cond	The condition to be passed in order for a
+ *              word to be counted. 
+ *              A function which is supplied a string as an
+ *              a boolean specifying whether the supplied
+ *              string is to be counted.
+ *
+ * \returns		A map containing unique words as keys and their
+ *              total number of occurrences as the associated
+ *              values.
+ */
+unordered_map<string, int> countUniqueWords(const vector<string>& words, const bool cond(string)) {
+	unordered_map<string, int> count;
+
+	// Iterates over each value in the specified vector \p words.
+	for (auto i = words.begin(); i != words.end(); ++i) {
+		string word = *i;
+		unordered_map<string, int>::iterator m_iter = count.find(*i);
```

Comment:
You seem to be case sensitive and you are not ignoring the punctuation. This needs to be added.